school_transport/models/transport.py

```diff
@@ -0,0 +1,623 @@
+        for rec in self:
+            delta = rec.end_date - rec.start_date
+            if rec.start_date > rec.end_date:
+                raise ValidationError(
+                    _(
+                        """
+                                Start date should be less than end date!"""
+                    )
+                )
+            if delta.days < 30:
+                raise ValidationError(_("Enter duration of month!"))
+
+    def unlink(self):
+        """Inherited method to check state at record deletion"""
+        for rec in self:
+            if rec.state == "open":
+                raise ValidationError(
+                    _(
+                        """
+            You can delete record in draft state or cancel state only!"""
+                    )
+                )
+        return super(StudentTransports, self).unlink()
+
+
+class StudentStudent(models.Model):
+    _inherit = "student.student"
+    _description = "Student Information"
+
+    transport_ids = fields.Many2many(
+        "transport.participant",
+        "std_transport",
+        "trans_id",
+        "stud_id",
+        "Transport",
+        help="Student transportation",
+    )
+
+    def set_alumni(self):
+        """Override method to make record of student transport active false.
+
+        when student is set to alumni state.
+        """
+        for rec in self:
+            trans_student_rec = self.env["transport.participant"].search(
+                [("name", "=", rec.id)]
+            )
+            trans_regi_rec = self.env["transport.registration"].search(
+                [("part_name", "=", rec.id)]
+            )
+            if trans_regi_rec:
+                trans_regi_rec.write({"state": "cancel"})
+            if trans_student_rec:
+                trans_student_rec.active = False
+        return super(StudentStudent, self).set_alumni()
+
+
+class TransportRegistration(models.Model):
+    """for registration."""
+
+    _name = "transport.registration"
+    _description = "Transport Registration"
+
+    @api.depends("state")
+    def _compute_get_user_groups(self):
+        """Method to compute transport user boolean field"""
+        transport_user_group = self.env.ref(
+            "school_transport.group_transportation_user"
+        )
+        grps = [group.id for group in self.env.user.groups_id]
+        self.transport_user = False
+        if transport_user_group.id in grps:
+            self.transport_user = True
+
+    @api.depends("m_amount", "for_month")
+    def _compute_transport_fees(self):
+        """Method to compute transport fees"""
+        self.transport_fees = self.m_amount * self.for_month
+
+    name = fields.Many2one(
+        "student.transport",
+        "Transport Root Name",
+        domain=[("state", "=", "open")],
+        required=True,
+        help="Enter transport root name",
+    )
+    part_name = fields.Many2one(
+        "student.student", "Student Name", required=True, help="Student Name"
+    )
```

Comment:
Did not understand the reason of the field name part_name. Here field name should be student_id,